android/README.md

```diff
@@ -0,0 +1,177 @@
+
+Please note that file upload will only work if, along with the above, you have also defined `publicKey`, `urlEndpoint`, and `authenticationEndpoint` variables in `app/config/Imagekit.js`.
+
+Install packages:
+
+```bash
+npm install
+```
+
+## Run the Node.js server
+
+```
+npm start
+```
+
+Node server will run at `http://localhost:8080`.
+
+
+### Important Note
+For ease for loading file, `app/src/main/java/io/imagekit/imagekitdemo/FileUtils.kt` can be imported into your project and used like it is demonstrated in `app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt`.
+
+# Demo Features Code Samples
+
+## Setting up ImageKit iOS SDK
+#### **`AppDelegate.swift`**
+``` swift
+import com.imagekit.android.ImageKit
+import com.imagekit.android.entity.TransformationPosition
+
+class MainActivity : AppCompatActivity() {
+  ...
+  override fun onCreate(savedInstanceState: Bundle?) {
+        ...
+
+        ImageKit.init(
+            context = applicationContext,
+            urlEndpoint = "https://ik.imagekit.io/YOUR_IMAGEKIT_ID",
+            transformationPosition = TransformationPosition.PATH,
+        )
+        ...
+  }
+  ...
+}
+```
+
+## Rendering images
+
+```swift
+var url = ImageKit.getInstance().url("plant.jpeg", "https://ik.imagekit.io/demo/img").create()
+
+```
+
+## Setting authentication context for the SDK
+#### **`AppDelegate.swift`**
+``` swift
+import com.imagekit.android.ImageKit
+import com.imagekit.android.entity.TransformationPosition
+
+class MainActivity : AppCompatActivity() {
+  ...
+  override fun onCreate(savedInstanceState: Bundle?) {
+        ...
+
+        ImageKit.init(
+            context = applicationContext,
+            publicKey = "YOUR_IMAGEKIT_PUBLIC_KEY",
+            urlEndpoint = "https://ik.imagekit.io/YOUR_IMAGEKIT_ID",
+            transformationPosition = TransformationPosition.PATH,
+            authenticationEndpoint = "http://localhost:8080/auth"
+        )
+        ...
+  }
+  ...
+}
+```
+
+
+## Applying common image manipulations
+```swift 
+let url = ImageKit.getInstance().url("plant.jpeg", "https://ik.imagekit.io/demo/img")
+            .width(300)
+            .height(200)
+            .cropMode(CropMode.PAD_RESIZE)
+            .background("F3F3F3")
+            .create()
+```
+
+## Adding overlays to images
+```swift 
+let url = ImageKit.getInstance()
+            .url(
+                src = "https://ik.imagekit.io/demo/medium_cafe_B1iTdD0C.jpg",
+                transformationPosition = TransformationPosition.PATH
+            )
+            .overlayImage("logo-white_SJwqB4Nfe.png")
+            .overlayX(10)
+            .overlayY(20)
+            .create()
+```
+
+## Client-side file uploading
+```swift
+ImageKit.getInstance().uploader().upload(
+    file = file!!
+    , fileName = file!!.name
+    , useUniqueFilename = true
+    , tags = arrayOf("nice", "copy", "books")
+    , folder = "/dummy/folder/"
+    , imageKitCallback = this
+)
+```
```

Comment:
These should be Android docs, not iOS

Reply:
Fixed the docs and also added code examples for adaptive streaming URL and 3rd party image library extensions